src/brogue/Recordings.c

```diff
@@ -1208,11 +1212,12 @@ void loadSavedGame() {
 
             executeEvent(&theEvent);
 
-            if (!(recordingLocation % progressBarInterval) && !rogue.playbackOOS) {
+            if (recordingLocation / progressBarInterval != previousRecordingLocation / progressBarInterval && !rogue.playbackOOS) {
```

Comment:
I think the way it was, even in 1.7.4, was that the loading bar would occasionally reload in chunks. It mods the length of the playback file with the current reading location, which goes up by 5 the mass majority of the time and would skip progressBarInterval every turn if it was divisible by 5. Then every 100 turns or so it would re-sync up and draw the loading bar. In advanceToLocation, it mods the total turn count with the current turn, which avoids this. 

What I have here checks if recordingLocation/progressBarInterval changes value every turn, but I don't know why the two loading bars have different implementations in the first place.

Reply:
I think it could be because the length in turns of a recording/save is not known, whereas advanceToLocation works on turns directly so can use them